packages/ra-ui-materialui/src/list/filter/FilterButton.tsx

```diff
@@ -182,15 +182,10 @@ export const FilterButton = (props: FilterButtonProps) => {
                     (filterElement: JSX.Element, index) => (
                         <FilterButtonMenuItem
                             key={filterElement.props.source}
-                            filter={{
-                                ...filterElement,
-                                props: {
-                                    ...filterElement.props,
-                                    applied: appliedFilters.includes(
-                                        filterElement.props.source
-                                    ),
-                                },
-                            }}
+                            filter={filterElement}
+                            displayed={
+                                !!displayedFilters[filterElement.props.source]
```

Comment:
`appliedFilters` is not needed anymore, right?

Reply:
right! I'll fix it.